src/server/editorServices.ts

```diff
@@ -1406,47 +1623,101 @@ namespace ts.server {
          */
         reloadProjects() {
             this.logger.info("reload projects.");
+            this.reloadConfiguredProjectForFiles(this.openFiles, /*delayReload*/ false);
+            this.refreshInferredProjects();
+        }
+
+        delayReloadConfiguredProjectForFiles(openFilesMap: Map<boolean>, ignoreIfNotInferredProjectRoot: boolean) {
+            // Get open files to reload projects for
+            const openFiles = flatMapIter(openFilesMap.keys(), path => {

```

Comment:
Since the callback never returns an array, it would help to add a `mapDefinedIter` helper for this purpose.
If `getScriptInfoForPath` intended to always return a defined value? It looks like in a few locations it is, but sometimes it isn't. Might help to rename it to `tryGetScriptInfoForPath`, and make `getScriptInfoForPath` call that and assert that it got the ScriptInfo.